agentMET4FOF_tutorials/tutorial_4_metrological_agents_and_streams.py

```diff
@@ -0,0 +1,87 @@
+import time
+from typing import Dict
+
+import numpy as np
+from time_series_metadata.scheme import MetaData
+
+from agentMET4FOF.agents import AgentNetwork
+from agentMET4FOF.metrological_agents import MetrologicalAgent, MetrologicalMonitorAgent
+from agentMET4FOF.metrological_streams import MetrologicalSineGenerator
+
+
+class MetrologicalSineGeneratorAgent(MetrologicalAgent):
+    """An agent streaming a sine signal
+
+    Takes samples from the :py:mod:`SineGenerator` and pushes them sample by sample
+    to connected agents via its output channel.
+    """
+
+    # The datatype of the stream will be SineGenerator.
+    _sine_stream: MetrologicalSineGenerator
+
+    def init_parameters(self, signal: MetrologicalSineGenerator = MetrologicalSineGenerator(), **kwargs):
+        """Initialize the input data
+
+        Initialize the input data stream as an instance of the
+        :py:mod:`SineGenerator` class
+
+        Parameters
+        ----------
+        signal : Signal
+            the underlying signal for the generator
+        """
+        self._sine_stream = signal
+        super().init_parameters()
+        self.set_output_data(channel="default", metadata=self._sine_stream.metadata)
+
+    def agent_loop(self):
+        """Model the agent's behaviour
+
+        On state *Running* the agent will extract sample by sample the input data
+        streams content and push it via invoking
+        :py:method:`AgentMET4FOF.send_output`.
+        """
+        if self.current_state == "Running":
+            self.set_output_data(
+                channel="default", data=[self._sine_stream.next_sample()]
+            )
+            super().agent_loop()
+
+    @property
+    def metadata(self) -> Dict:
+        return self._sine_stream.metadata.metadata
+
+
+def main():
```

Comment:
I think we should rather name this less generic like so:
```suggestion
def demonstrate_metrological_stream():
```